src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -74,7 +77,7 @@ public class Session {
             _pipe = new MessagePipe(messageLogger);
         }
 
-        public void StartHost(SecureString password, string profilePath, ILogger outputLogger, LogVerbosity verbosity) {
+        public void StartHost(ClaimsPrincipal cp, string profilePath, ILogger outputLogger, LogVerbosity verbosity) {
```

Comment:
Also, it's not clear how it should interact with `User` - i.e. what happens if the caller passes a principal for a user different from `IIdentity` that was passed to the session constructor.